lib/content-services/src/lib/node-comments/services/node-comments.service.spec.ts

```diff
@@ -81,5 +84,44 @@ describe('NodeCommentsService', () => {
                 responseText: JSON.stringify(fakeContentComments)
             });
         });
+
+        it('should cache the avatar URL when new comment is processed', (done) => {
+            spyOn(contentService, 'getContentUrl').and.returnValue('fake-url');
+
+            service.add('999', fakeContentComment.entry.content).subscribe((result: CommentModel) => {
+                expect(contentService.getContentUrl).toHaveBeenCalledWith('123-123-123');
+                expect(service.getUserImage('fake-email@dom.com')).toBe('fake-url');
+                expect(result.createdBy.id).toBe('fake-email@dom.com');
+                done();
+            });
+
+            jasmine.Ajax.requests.mostRecent().respondWith({
```

Comment:
If you want to mock out this.commentsApi.createComment then you should not use jasmine.Ajax for that. We should not touch anything outside file which we test so you should instead spyOn createComment function on commentsApi and then you can mock out result of that function